lib/Model/SitemapPage.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Website\Model;
+
+use DateTimeImmutable;
+use Doctrine\SkeletonMapper\Hydrator\HydratableInterface;
+use Doctrine\SkeletonMapper\Mapping\ClassMetadataInterface;
+use Doctrine\SkeletonMapper\Mapping\LoadMetadataInterface;
+use Doctrine\SkeletonMapper\ObjectManagerInterface;
+
+class SitemapPage implements HydratableInterface, LoadMetadataInterface
+{
+    /** @var string */
+    private $url;
+
+    /** @var DateTimeImmutable */
+    private $date;
+
+    public function __construct(string $url, DateTimeImmutable $date)
+    {
+        $this->url  = $url;
+        $this->date = $date;
+    }
+
+    public static function loadMetadata(ClassMetadataInterface $metadata) : void
+    {
+        $metadata->setIdentifier(['url']);
+    }
+
+    /**
+     * @param mixed[] $sitemapPage
+     */
+    public function hydrate(array $sitemapPage, ObjectManagerInterface $objectManager) : void
```

Comment:
Types for the properties should be handled in this method or we can get an invalid object state.